pubsub/subscriptions/create_dead_letter.go

```diff
@@ -0,0 +1,55 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package subscriptions
+
+// [START pubsub_create_subscription_dead_letter]
+import (
+	"context"
+	"fmt"
+	"io"
+	"time"
+
+	"cloud.google.com/go/pubsub"
+)
+
+func createSubWithDeadLetter(w io.Writer, projectID, subID string, topicID string, deadLetterTopicName string) error {
+	// projectID := "my-project-id"
+	// subID := "my-sub"
+	// topicID := "my-topic"
+	// deadLetterTopicName := "projects/my-project/topics/my-dead-letter-topic"
```

Comment:
`deadLetterTopicID := "my-dead-letter-topic"` is better here for consistency reasons.

Reply:
We have to use the fully qualified name here since the call expects that. The same applies for `update` as well.
I think this is slightly more clear than assuming the subscription and dead letter topic are in the same project.